bugsnag-spring/src/main/java/com/bugsnag/MvcConfiguration.java

```diff
@@ -1,36 +1,48 @@
 package com.bugsnag;
 
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Conditional;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.HandlerExceptionResolver;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
 
+import java.util.List;
 import javax.annotation.PostConstruct;
 
 /**
  * If spring-webmvc is loaded, add configuration for reporting unhandled exceptions.
  */
 @Configuration
 @Conditional(SpringWebMvcLoadedCondition.class)
-class MvcConfiguration {
+class MvcConfiguration extends WebMvcConfigurerAdapter {
 
     @Autowired
     private Bugsnag bugsnag;
 
-    /**
-     * Register an exception resolver to send unhandled reports to Bugsnag
-     * for uncaught exceptions thrown from request handlers.
-     */
-    @Bean
-    BugsnagMvcExceptionHandler bugsnagHandlerExceptionResolver() {
-        return new BugsnagMvcExceptionHandler(bugsnag);
-    }
-
     /**
      * Add a callback to assign specified severities for some Spring exceptions.
      */
     @PostConstruct
     void addExceptionClassCallback() {
         bugsnag.addCallback(new ExceptionClassCallback());
     }
+
+    /**
+     * Normally, the exceptionResolvers contain the following resolvers in this order:
+     * - {@link org.springframework.web.servlet.mvc.method.annotation.ExceptionHandlerExceptionResolver}
+     * - {@link org.springframework.web.servlet.mvc.annotation.ResponseStatusExceptionResolver}
+     * - {@link org.springframework.web.servlet.mvc.support.DefaultHandlerExceptionResolver}
+     * <p>
+     * The first two handlers handle exceptions in an application-specific manner.
+     * (either by @{@link org.springframework.web.bind.annotation.ExceptionHandler}
+     * or @{@link org.springframework.web.bind.annotation.ResponseStatus})
+     * <p>
+     * Therefore, exceptions that are handled by these handlers should not be handled by Bugsnag.
+     * Only unhandled exceptions shall be sent to Bugsnag.
+     */
+    @Override
+    public void extendHandlerExceptionResolvers(List<HandlerExceptionResolver> exceptionResolvers) {
+        final int position = exceptionResolvers.isEmpty() ? 0 : exceptionResolvers.size() - 1;
```

Comment:
As far as I can tell the default exception resolvers are populated [here](https://github.com/spring-projects/spring-framework/blob/main/spring-webmvc/src/main/java/org/springframework/web/servlet/config/annotation/WebMvcConfigurationSupport.java#L1075).

My concern with this approach is that we don't have any control over the contents of this list. If the order or content of the default exception handlers is changed in the spring libraries then it could affect the way that the Bugsnag library works. If it does then we might end up in the situation of trying to support multiple versions of this mechanism as we can't guarantee what versions of the spring framework users are using.

Although there are problems with the current implementation, I believe that it is more resilient to change in the framework, and it's still possible to exclude unwanted exceptions using the `beforeNotify` callback, or by specifying "Discard by error class" in the project settings.